infra/src/main/kotlin/fi/fta/geoviite/infra/tracklayout/LayoutDesign.kt

```diff
@@ -1,5 +1,20 @@
 package fi.fta.geoviite.infra.tracklayout
 
+import fi.fta.geoviite.infra.common.DomainId
 import fi.fta.geoviite.infra.common.IntId
+import fi.fta.geoviite.infra.geometry.PlanPhase
+import fi.fta.geoviite.infra.util.FreeText
+import java.time.Instant
+import java.time.LocalDate
 
-data class LayoutDesign(val id: IntId<LayoutDesign>)
+enum class DesignState {
+    ACTIVE, DELETED, COMPLETED
+}
+
+data class LayoutDesign(
+    val id: DomainId<LayoutDesign>,
+    val name: FreeText,
```

Comment:
Laittaisin tuolle nimellekin ehkä kuitenkin oman tyypin, koska FreeText sallii kaikenlaista rivinvaihdoista alkaen ja ilman mitään pituusrajoja. Koska tuo on säisäinen nimi, sille voi laittaa suht tiukatkin säännöt: <1000 merkkiä, alfanumeeriset, sulut, viivat, perus-punktuaatiot ja välilyönnit varmaan riittäisi. Sääntöjen löysentäminen myöhemmin on aina turvallista, mutta tiukennuksissa joutuu varmistamaan ettei olemassaoleva data tule haitaksi.